lib/chef/cookbook_loader.rb

```diff
@@ -66,17 +66,27 @@ def merged_cookbook_paths # for deprecation warnings
       merged_cookbook_paths
     end
 
+    def warn_about_cookbook_shadowing
+      unless merged_cookbooks.empty?
+        Chef::Log.deprecation "The cookbook(s): #{merged_cookbooks.join(', ')} exist in multiple places in your cookbook_path. " +
+          "A composite version has been compiled.  In a future version of Chef this behavior will be removed."
```

Comment:
hmmm, maybe we should throw in a Chef::Config conditional to remove shadow cookbooks now and have it default to enabled...

that'd really be the right way to do this, but not at 6pm on a friday before a 3 day weekend...